crates/lint/src/sol/info/named_struct_fields.rs

```diff
@@ -16,16 +16,58 @@ impl<'hir> LateLintPass<'hir> for NamedStructFields {
     fn check_expr(
         &mut self,
         ctx: &LintContext,
-        _hir: &'hir solar::sema::hir::Hir<'hir>,
+        hir: &'hir solar::sema::hir::Hir<'hir>,
         expr: &'hir solar::sema::hir::Expr<'hir>,
     ) {
-        if let ExprKind::Call(
-            Expr { kind: ExprKind::Ident([Res::Item(ItemId::Struct(_struct_id))]), .. },
-            CallArgs { kind: CallArgsKind::Unnamed(_args), .. },
+        let ExprKind::Call(
+            Expr { kind: ExprKind::Ident([Res::Item(ItemId::Struct(struct_id))]), span, .. },
+            CallArgs { kind: CallArgsKind::Unnamed(args), .. },
             _,
         ) = &expr.kind
-        {
+        else {
+            return;
+        };
+
+        let strukt = hir.strukt(*struct_id);
+        let fields = &strukt.fields;
+
+        // Basic sanity conditions for a consistent auto-fix
+        if fields.len() != args.len() || fields.is_empty() {
+            // Emit without suggestion
             ctx.emit(&NAMED_STRUCT_FIELDS, expr.span);
+            return;
         }
+
+        // Get struct name snippet and emit without suggestion if we can't get it
+        let Some(struct_name_snippet) = ctx.span_to_snippet(*span) else {
+            // Emit without suggestion if we can't get the struct name snippet
+            ctx.emit(&NAMED_STRUCT_FIELDS, expr.span);
+            return;
+        };
+
+        // Collect field names and corresponding argument source snippets
+        let mut field_assignments = Vec::new();
+        for (field_id, arg) in fields.iter().zip(args.iter()) {
+            let field = hir.variable(*field_id);
+            let field_name = field.name.map(|n| n.to_string()).unwrap_or_else(|| "?".to_string());
+
+            let Some(arg_snippet) = ctx.span_to_snippet(arg.span) else {
+                // Emit without suggestion if we can't get argument snippet
+                ctx.emit(&NAMED_STRUCT_FIELDS, expr.span);
+                return;
+            };
```

Comment:
```suggestion
            let field = hir.variable(*field_id);

            let Some((field_name, arg_snippet)) = ctx.span_to_snippet(arg.span).zip(field.name.map(|n| n.to_string()) else {
                // Emit without suggestion if we can't get argument snippet
                ctx.emit(&NAMED_STRUCT_FIELDS, expr.span);
                return;
            };
```

if the field name is undefined we want the fix-less diagnostic as well. we can use [`Option::zip`](https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.zip)